### DISCUSSION TITLE
Stability (uptime) of esp8266 version 2.3.8

### DISCUSSION THREAD

**Discussion by kardesken** [UtilitechAS/amsreader-firmware#839] [General]
I want to say that the stability of esp8266 have improved a lot for me.
(I don't know if this happened in version 2.3.7 or 2.3.8).

I'm using this software with an ESP-12F: ESP8266 module. And after the last update (2.3.8), my devise haven't rebooted a single time . It's currently at 9 days uptime.

Previously my device would reboot every one or two days. 
I have never had any other issues regarding the reboots. And if it wasn't for me checking the beautiful web interface and reading the "up time", I would never have known.

I'm using this device together with Home Assistant trough MQTT, and it's just great. 

Oh, and only last week I noticed the upgrade to the pricing that have been integrated with the "Configure price" menu/configuration. Nice!

So thank you for the great work and for sharing the software with us.

**Comment by ArnieO** [comment]
Thank you for your kind words! 
And thank you for feedback regarding stability of latest firmware revisions. 
The uptime counter is both a "curse" and a very useful tool to assess firmware stability. 

There are very few that actually notice anything if the device reboots from time to time - but it depends on several factors. For users in Norway and Sweden that are on HAN-NVE configured meters, a reboot messes up the realtime estimations until next passing of whole hour. The reason is that those meters only send accumulated energy [kWh] once per hour.

We have chased a reboot issue for quite some time, and think there is something going on outside of our control, maybe in one of the libraries we use (MQTT library is a "suspect"). Which means there could be an improvement if an underlying library has been updated.
